org.eclipse.jdt.ls.product/scripts/jdtls

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+import argparse
+import glob
+import os
+import platform
+import re
+import semver
+import subprocess
+import sys
+import defusedxml.ElementTree as ET
+from pathlib import Path
+
+def get_java_executable(validate_java_version):
+	java_executable = 'java'
+
+	if 'JAVA_HOME' in os.environ:
+		java_exec_to_test = os.path.join(os.environ['JAVA_HOME'], 'bin', 'java')
+		if os.path.exists(java_exec_to_test):
+			java_executable = java_exec_to_test
+
+	if validate_java_version:
+		out = subprocess.check_output(java_executable + ' -version', shell = True, stderr = subprocess.STDOUT, universal_newlines=True)
```

Comment:
I think I asked that before: why with `shell=True`? That only gives the potential for injections.

```suggestion
		out = subprocess.check_output([java_executable, '-version'], stderr = subprocess.STDOUT, universal_newlines=True)
```

Reply:
Sorry, I did not notice the suggestion last time. I'll cover it.